geometry_definitions.py

```diff
@@ -0,0 +1,160 @@
+    fuel_assembly_cell = openmc.Cell(region=-fuel_assembly, fill=fuel)
+    fuel_assembly_cladding_cell = openmc.Cell(
+        region=+fuel_assembly & -fuel_assembly_cladding, fill=clad)
+
+    outer_lattice_universe = openmc.Universe(cells=[fuel_assembly_cell])
+
+    # OpenMC Fuel Lattice
+    fuel_lattice = openmc.HexLattice()
+    fuel_lattice.orientation = "x"
+    fuel_lattice.outer = outer_lattice_universe
+    fuel_lattice.pitch = (propellant_channel_pitch,)
+    fuel_lattice.universes = [[propellant_channel] *
+                              12, [propellant_channel]*6, [propellant_channel]]
+    fuel_lattice.center = (0.0, 0.0)
+
+    # Full Fuel Asembly
+    fuel_assembly_lattice_cell = openmc.Cell(
+        region=-fuel_assembly, fill=fuel_lattice)
+    fuel_assembly_universe = openmc.Universe(cells=[fuel_assembly_cladding_cell,
+                                                    fuel_assembly_lattice_cell])
+
+    return fuel_assembly_universe
+
+
+def tie_tube(hydrogen_inner, hydrogen_outer, inconel, ZrH, ZrC):
+    #THIS IS A GUESS BASED ON THE FUEL ELEMENT
+    assembly_cladding_thickness = 0.005
+    
+    inner_hydrogen_outer_radius = 0.20955
+    propellant_channel_pitch = 0.4089
+    inner_tie_tube_outer_radius = 0.26035
+    first_gap_outer_radius = 0.26670
+    moderator_outer_radius = 0.58420
+    second_gap_outer_radius = 0.67818
+    outer_tie_tube_outer_radius = 0.69850
+    third_gap_outer_radius = 0.70485
+    insulator_outer_radius = 0.80645
+    fourth_gap_outer_radius = 0.81280
+    flat_to_flat = 1.905
+    flat_to_flat_inner = flat_to_flat-2*assembly_cladding_thickness
+    assembly_edge_length = 0.5*flat_to_flat/np.cos(np.deg2rad(30))
+    inner_edge_length = 0.5*flat_to_flat_inner/np.cos(np.deg2rad(30))
+
+    # OpenMC Geometry
+    inner_hydrogen = openmc.ZCylinder(r=inner_hydrogen_outer_radius)
+    inner_tie_tube = openmc.ZCylinder(r=inner_tie_tube_outer_radius)
+    first_gap = openmc.ZCylinder(r=first_gap_outer_radius)
+    moderator_tube = openmc.ZCylinder(r=moderator_outer_radius)
+    second_gap = openmc.ZCylinder(r=second_gap_outer_radius)
+    outer_tie_tube = openmc.ZCylinder(r=outer_tie_tube_outer_radius)
+    third_gap = openmc.ZCylinder(r=third_gap_outer_radius)
+    insulator = openmc.ZCylinder(r=insulator_outer_radius)
+    fourth_gap = openmc.ZCylinder(r=fourth_gap_outer_radius)
+    tie_tube_assembly = openmc.model.HexagonalPrism(orientation='x',
+                                                    edge_length=inner_edge_length)
+    tie_tube_assembly_cladding = openmc.model.HexagonalPrism(orientation='x',
+                                                             edge_length=assembly_edge_length)
+
+    # OpenMC Cells and Universes
+    inner_hydrogen_cell = openmc.Cell(region=-inner_hydrogen, fill=hydrogen_inner)
+    inner_tie_tube_cell = openmc.Cell(
+        region=+inner_hydrogen & - inner_tie_tube, fill=inconel)
+    first_gap_cell = openmc.Cell(region=+inner_tie_tube & -first_gap)
```

Comment:
All four gaps should be filled with hydrogen according to that table I sent

Reply:
filling all gaps with `hydrogen_outer`, doubtful those very thin layers will make much of an impact but we should explore that to be sure